### DISCUSSION TITLE
Incorporation of the Cistercian version: how?

### DISCUSSION THREAD

**Discussion by Augustinus-Altovadensis** [DivinumOfficium/divinum-officium#4053] [Developers corner]
Laudetur Jesus Christus!

I'm almost ready to start working on the Cistercian version, as we spoke before.

My question is: 
1. how do I do it? Do I simply create a pull request and enter all at once? 
2. Or should I first upload the non-program files that won't affect anything and then the .pl files? 
3. Can I in practice upload my changes to the code, when it's changing all the time? Or shall I rather upload the changes for each day I'm working on it and do it gradually? There is about 100 addition to the code to add the required functionality for the Cistercian Office (while keeping the rest intact, of course), so it will take a while...

Thanks for the reply, God bless!

**Comment by FAJ-Munich** [comment]
In æternum!

After checking the rubrics you have kindly provided early August, it seems to me that the Cistercian 1951 is most similar to the Benedictine 1617 version currently in construction as well. The partial adoption of the 1957 rules by the Abby you mentioned, of course changes things ever so shortly. 
My feeling right now is, we can exploit a lot of existing code in a clever way if incorporating the required code changes and database carefully. 
Which of your branches is the definitive source you would like to start the incorporation exercise?

**Reply by Augustinus-Altovadensis** [reply]
I'm using only the `master` branch in my fork, so that's the one. 

However, do you think I should use my fork to send in the pull requests, or is it better to start from a new one, or directly at DivinumOfficium master or other branch?

Yes, the Cistercian rubrics are very conservative. Apart from the calendar and classes of feasts, there are some changes like feasts of iij. Lect. et M. have ferial Psalms after the Cistercians' application of Divino Afflatu, but still use the Antiphon ad Laudes from the Common, which is something that I wasn't able to find anywhere (in the older version).

**Reply by FAJ-Munich** [reply]
I have cloned your branch locally now to browse through your code changes and see how to incorporate them smoothly.

**Reply by Augustinus-Altovadensis** [reply]
Very good. It's not based on the newest version, but I guess you already know that.

Also, the edits in the `concurrence()` and `occurrence()` functions are largely hit and miss. The aim is to show all the commemorations of feasts of rank lower than 7, and almost all commemorations at feasts of rank equal to 7, with the only exception being commemorations of rank = 1, that get commemorations in Lauds only. So all the Ferias in Advent and Lent need to be commemorated even in Serm. maj. (rank = 7) feasts. This still doesn't work. The rest is pretty much fine.

**Reply by FAJ-Munich** [reply]
After re-reading the Rubrics as per the 1951 book, I see the following differences to Monastic 1617:

- Festum sermonis is like duplex I. cl. in occurence but only like duplex II. cl. in concurrence
- Dominica I. cl. has Vesp. a capitulo in concurrence even against duarum missarum
- Feria major in Adventu usque ad 16 dec. gives way to iii. lect. et. M. and S. Maria in Sabbato
- Suffrages are said even in some Festis duarum Missarum minoribus which is not the case for Duplex minus

All in all, I don't think it is going to pose to much trouble to "encode" this in the existing rankings as step 1. 

After cloning your repository, however, I was wondering why, in your version, Matins only renders one nocturn and finishes directly  with the oratio?  And do you have a rough estimate how different the actual texts are to the Benedictine version?

**Reply by Augustinus-Altovadensis** [reply]
If those are the only differences, it's really not that bad. It's more complicated than that. 

- Festum sermonis is major and minor, with only Serm. maj. having the power to suppress a simple Commemoration in first Vespers, but only that, i.e. anything with rank higher than that must be displayed.
- Dominica I. cl. has ferial Psalms in first Vespers, but otherwise it trumps everything else, including Serm. maj. (which gets transferred, of course).
- Yes, that's true. And it must be commemorated, e.g. on 8th Dec., even though it's a Serm. maj.
- True. However, in Vyšší Brod, we don't say the Suffrages in any feast higher than xij. Lect. et M. And the brothers, after some snide remarks from the Abbot Praeses, have decided to partially implement the provision from 1957, that when considering the Suffrages, the total number of Commemorations (+ Suffrages) shouldn't exceed 3. 

I haven't basically touched the Matins, so their current state is a result of chance and various changes. In the future, I'd like to make at least the Ferias and common texts up to date, but currently it's not a priority, as I nobody who uses my fork doesn't pray Matins, and supplying all the texts would be so much work, although I have all the reading already scanned and translated for the brothers... Anyway, all the Matins have at least two Nocturns with 6 Psalms each, Festum iij. Lect. et M. has 3 Lessons after the first Nocturn, and every higher feast has 12 Lessons, 4 after each Nocturn, and the 3rd Nocturn has the OT Canticles... This all can and will be implemented, but first I want to get up to date with the Diurnale, both coding and Calendar.

And what is the difference between Benedictines and us in Matins? Given that the differences between the Roman office (if the feast is the same) are minor, it wouldn't be that different, at least looking at the Lessons. I imagine there will be differences in Responsories though...

**Reply by Augustinus-Altovadensis** [reply]
So what's the next step? Just to create a pull request when I'm ready?

**Reply by FAJ-Munich** [reply]
> If those are the only differences, it's really not that bad. It's more complicated than that. 

To the extent possible, to keep the changes to the code as minor as can be, let's logically separate the 1951 pure book rules and the 1957/2024 changes as applied by the brothers. 

> 
> 
> 
> - Festum sermonis is major and minor, with only Serm. maj. having the power to suppress a simple Commemoration in first Vespers, but only that, i.e. anything with rank higher than that must be displayed.

According to your scan, the distinction between major and minor is irrelevant for commemorations. I guess that's a 1957 change then?

> 
> - Dominica I. cl. has ferial Psalms in first Vespers, but otherwise it trumps everything else, including Serm. maj. (which gets transferred, of course).

That's a consequence of Divino Afflatu: 1957 only then!?

> 
> - Yes, that's true. And it must be commemorated, e.g. on 8th Dec., even though it's a Serm. maj.
> 
> - True. However, in Vyšší Brod, we don't say the Suffrages in any feast higher than xij. Lect. et M. And the brothers, after some snide remarks from the Abbot Praeses, have decided to partially implement the provision from 1957, that when considering the Suffrages, the total number of Commemorations (+ Suffrages) shouldn't exceed 3. 

Also a 1951/1957 difference then. 

> 
> 
> 
> I haven't basically touched the Matins, so their current state is a result of chance and various changes. In the future, I'd like to make at least the Ferias and common texts up to date, but currently it's not a priority, as I nobody who uses my fork doesn't pray Matins, and supplying all the texts would be so much work, although I have all the reading already scanned and translated for the brothers... Anyway, all the Matins have at least two Nocturns with 6 Psalms each, Festum iij. Lect. et M. has 3 Lessons after the first Nocturn, and every higher feast has 12 Lessons, 4 after each Nocturn, and the 3rd Nocturn has the OT Canticles... This all can and will be implemented, but first I want to get up to date with the Diurnale, both coding and Calendar.
> 

Using the Monastic routines as they are now, that shouldn't be too much of a hassle. 

> 
> 
> And what is the difference between Benedictines and us in Matins? Given that the differences between the Roman office (if the feast is the same) are minor, it wouldn't be that different, at least looking at the Lessons. I imagine there will be differences in Responsories though...

We can cover this with "magic comments"

**Reply by FAJ-Munich** [reply]
> So what's the next step? Just to create a pull request when I'm ready?

I am willing to incorporate and build from your existing branch. After first review, it is easier to incorporate it bit by bit into the new logic then a rebase or merge in one go can deliver. That way we can also exploit recent developments in the code. 

To keep in the spirit of late Laszlo Kiss, being able to compare versions is a plus. In addition, I am personally excited about the pure 1951 version you have shared the rubrics with me. They are almost exactly what I have dreamed of for my personal side branch and seeing it coincides with some rubrics actually appropriated once is just a blessing. Therefore, let's develop both versions side by side: 1951 pure and 1957/2024 Altovado. 

Timewise, I am almost through with a revision of the Commons which I like to finish and PR before jumping into this. I hope the weekend will be fruitful! Then I am up to jump right into this. 

One more thing that would be helpful: you mentioned more scans you have made for the brothers. If you could add that to your Dropbox would be great as it helps to be able to consult the source whenever in doubt.

**Reply by Augustinus-Altovadensis** [reply]
Of course, given how the Monastic version is structured, I completely agree with doing both pure 1951 and Altovadum version.

Difference between Serm. maj. and min. Yes, you're right, the rubrics don't differentiate much, it's all in the _nisi aliter notetur_. So it's not really a 1957 change, just something that comes up on specific occasions. For example, I have never seen a Serm. min. deleting first Vesper Commemoration of any lower feast, while in Serm. maj. this is a norm. 

The _Dominicæ majores_ (first class) are much older, the 1957 supplement only expanded their number, which is something that is not yet fixed in Vyšší Brod, so we can safely ignore it.

Yes, the Suffrages are based on 1957, only there it says that the complete number of Collects need to be 3, i.e. one main and two Commemorations, which is something we don't do in Vyšší Brod, only when it comes to Suffrages. Again, this is something that can be safely ignored, especially in the 1951 version.

Good. Yes, I have been using the "magic comment" `rubrica cisterciensis` quite a lot. It doesn't always work (like in ferial Versicles in `Major/Minor Special.txt`), but mostly it does.

**Reply by Augustinus-Altovadensis** [reply]
> > So what's the next step? Just to create a pull request when I'm ready?
> 
> I am willing to incorporate and build from your existing branch. After first review, it is easier to incorporate it bit by bit into the new logic then a rebase or merge in one go can deliver. That way we can also exploit recent developments in the code.
> 

Very good. That means I don't need to do anything (in the code) and you will take care of that? 
I will continue working on the Calendar and proper texts, of course.

> To keep in the spirit of late Laszlo Kiss, being able to compare versions is a plus. In addition, I am personally excited about the pure 1951 version you have shared the rubrics with me. They are almost exactly what I have dreamed of for my personal side branch and seeing it coincides with some rubrics actually appropriated once is just a blessing. Therefore, let's develop both versions side by side: 1951 pure and 1957/2024 Altovado.
> 
Sure, I like this idea.

> Timewise, I am almost through with a revision of the Commons which I like to finish and PR before jumping into this. I hope the weekend will be fruitful! Then I am up to jump right into this.
> 
Nice!
> One more thing that would be helpful: you mentioned more scans you have made for the brothers. If you could add that to your Dropbox would be great as it helps to be able to consult the source whenever in doubt.

Well, what I have scanned is mostly the Matins, which are quite in a disorder. I can share them with you, of course, but it's a mess. 
However, there are two more books that I can share:
Breviarium Cisterciense, pars vernalis:
https://www.dropbox.com/scl/fi/lym2fa90jy60pjfrofx15/Breviarium-Cisterciense-Vernalis.pdf?rlkey=b3g9junhjc554wkb35n3zaye4&dl=0

And Rituale cisterciense from 1892. Not everything is still valid, but it's a nice source for Cistercian life in common.
https://www.dropbox.com/scl/fi/s034097t9c6tqud1ik995/Rituale-cisterciense-Lerins-1892-images.pdf?rlkey=cvm8t2zh9es2a4rjkscatijkn&dl=0

**Reply by FAJ-Munich** [reply]
> Very good. That means I don't need to do anything (in the code) and you will take care of that? 

That's my plan. 


> Well, what I have scanned is mostly the Matins, which are quite in a disorder. I can share them with you, of course, but it's a mess. 
> 

No worries. What really does help are scans of the books as below. 

> However, there are two more books that I can share:
> 
> Breviarium Cisterciense, pars vernalis:
> 
> https://www.dropbox.com/scl/fi/lym2fa90jy60pjfrofx15/Breviarium-Cisterciense-Vernalis.pdf?rlkey=b3g9junhjc554wkb35n3zaye4&dl=0

That's perfect and helps a lot to get going. Of course, the other three parts, if you have them, would be great, too. But for now, this is already a great start.

**Reply by Augustinus-Altovadensis** [reply]
Very good, thank you! 

Unfortunately, the Pars Vernalis is the only one that we have scanned. As you can imagine, it's a lot of work, so nobody is eager to scan the rest. That's also why I shared with you the two Antiphonaries and the Psalter. Apart from the Lessons for Matutinum, everything is there. 

I don't want to promise you scanning any of the Breviaries in their entirety, but if you'd like to have some parts, I can do it separately. I mean like a feast or two, not like the entire Santorale or an entire month, of course. Currently, I need to finish a new book of chants for the FSSP, perpetual liturgical Ordo App for the brothers, a Latin-Czech version of the Diurnale Cisterciense, and I simply have no time to do the scanning...

**Reply by FAJ-Munich** [reply]
> I don't want to promise you scanning any of the Breviaries in their entirety, ...

no worries; I can imagine just how much effort that would require

**Reply by Augustinus-Altovadensis** [reply]
Thank you :-)

**Reply by FAJ-Munich** [reply]
Laudetur Jesus Christus!

I have started with the first minor parts of the implementation on the following branch:
https://github.com/FAJ-Munich/divinum-officium/tree/cist-develop

I am going to use "rebase" and "force push" for the time being and for as long as possible but it should give you an option to track progress. Once I have implemented more (now there is almost nothing), please feel free to post "issues" on the project I have created as part of my fork to which I have invited you as a collaborator. If you want, how about you create another branch off the upstream/master in your fork to which I can PR my changes going forward for you to test it out on your local site as well? 

And, finally, once question I have been thinking about on the 1951/1957 changes:

- [ ] In 1951, even xij. lect. and M. has the right of transferal. How about under 1957/Altovado rules? This is especially important given that Sundays are elevated to outrank MM. maj. just like Divino Afflatu has ordered.

**Comment by Augustinus-Altovadensis** [comment]
In ætérnum!

Very good, yes, I have just created a branch from upstream/master, called `cist-develop`, where you can push your changes for testing.

Concerning the current Altovadum rules, it differs according to the Season. The green Sunday is elevated to outrank the **MM. min.** (not MM. maj.) and if such or lower feast falls on such a Sunday, it gets commemorated and not transferred. I suppose this is mostly a practical rule, as transferrals especially during the Summer months might get quite problematic, and no one wants a feast transferred over two or three weeks... However, during Advent and Lent (i.e. all Sundays are Dominicæ majores j. vel ij. classis), the feasts are transferred. The exact wording is still under development, as the brothers keep studying the rubrics in various places, so it might change a bit in the future...

**Comment by FAJ-Munich** [comment]
LJCh!

Here's the current status for December on my branch: So far, I have only worked on incorporating Cisterciense Occurrence and Concurrence rules. Not on the actual texts which are still displayed as if it was Benedictine Monastic. I think I hvae a good "minimal working example" now to continue with the Psalterium next:
![2024-12_Cist](https://github.com/user-attachments/assets/b2c31631-83db-45ee-a0ca-71784147ace4)

**Reply by Augustinus-Altovadensis** [reply]
Nice work! So far it looks good! Of course, St. Thomas of Canterbury is in both 1951 and Altovadum on 29.12., and on 30.12. it's Die vj. infra Oct. Nat., but otherwise it's good. I hope the Comm. of Feria on St. Nicolas as well as Immaculate Conception will be displayed in both Vespers and Lauds. This is something that didn't really work for me in all the Serm. maj. feasts. 

Also, it's not necessary to do it now, but wherever you look in the Antiphonary, Breviary, Missal, Gradual etc., the feasts are labeled as Serm. maj., Serm. min., MM. maj., MM. min., xij. Lect. et M., iij. Lect. et M., Commemoratio et M. and Commemoratio. And I think it would be prudent to keep it here too. 

BTW, if I can help in any way, let me know.

**Reply by FAJ-Munich** [reply]
> Nice work! So far it looks good! Of course, St. Thomas of Canterbury is in both 1951 and Altovadum on 29.12., and on 30.12. it's Die vj. infra Oct. Nat., but otherwise it's good. 

Of course! This year, however, 29 Dec falls on a Sunday which, as I have understood from you supersedes St. Thomas being MM. min. in Altovadum rules? Then, St Thomas still has the right of transferral, hasn't it? For St. Sylvester whose feast fell on a Sunday in 2023, it would even be necessary to transfer it to January 2 or 3 depending on the date for Ss. Nominis although this just doesn't feel right. 
Typically, the Pars Hiemalis contains some lengthy rubrics ahead of S. Stephanus and Dominica infra 8am Nativitatis. A picture of these would be great.

>I hope the Comm. of Feria on St. Nicolas as well as Immaculate Conception will be displayed in both Vespers and Lauds. This is something that didn't really work for me in all the Serm. maj. feasts. 
> 

Haven't checked yet but will be working in the end. 

> 
> 
> Also, it's not necessary to do it now, but wherever you look in the Antiphonary, Breviary, Missal, Gradual etc., the feasts are labeled as Serm. maj., Serm. min., MM. maj., MM. min., xij. Lect. et M., iij. Lect. et M., Commemoratio et M. and Commemoratio. And I think it would be prudent to keep it here too. 
> 

No problem. Can do. 


> 
> 
> BTW, if I can help in any way, let me know.

Am going to let you know once I have reached some stability in the code.

**Reply by Augustinus-Altovadensis** [reply]
> Of course! This year, however, 29 Dec falls on a Sunday which, as I have understood from you supersedes St. Thomas being MM. min. in Altovadum rules? Then, St Thomas still has the right of transferral, hasn't it? For St. Sylvester whose feast fell on a Sunday in 2023, it would even be necessary to transfer it to January 2 or 3 depending on the date for Ss. Nominis although this just doesn't feel right. Typically, the Pars Hiemalis contains some lengthy rubrics ahead of S. Stephanus and Dominica infra 8am Nativitatis. A picture of these would be great.

Oh my, you are right! Well, so far it has been valid only for the Green Sundays, because they get most often trumped by a number of feasts. Anyway, I will confirm it with the brothers, whether the Sunday in Christmas Octave has a precedence over St. Thomas More. In my opinion it shouldn't. And we certainly didn't transfer St. Sylvester last year, but by that time, this "rule" has been still in development... I will let you know.

Concerning lengthy rubrics... I have checked and there is nothing like that. The reason is quite simple, unless the rubrics have been messed with, the Christmas Octave goes as stated in the books. The only variable is the Dies vj., that may fall on Sunday or not. That's it. You may notice that unlike any other time of year, here everything is in the Temporale part, even the feast of Saints are there, instead of Sanctorale. 

BTW, I have found the 1885 Breviarium Cisterciense in Google Books. I have checked, the Christmas rubrics in the text are identical with the 1951 version, as is 99% of the Breviary.
This is Pars hiemalis:
https://www.google.de/books/edition/Breviarium_Cisterciense/fWYpAAAAYAAJ

You don't need the Vernalis, but it's there too:
https://www.google.de/books/edition/Breviarium_Cisterciense_Reformatum/7qspAAAAYAAJ

And Pars Æstivalis:
https://www.google.de/books/edition/Breviarium_Cisterciense_reformatum_juxta/Wzy9l_HIZcIC

Pars Autumnalis hasn't been scanned (yet?).

> No problem. Can do.
Thank you!

> > BTW, if I can help in any way, let me know.
> 
> Am going to let you know once I have reached some stability in the code.

Very good! BTW, I could not help to look some more into the changes, and I have one more wish:
in the magic comments in `web\cgi-bin\DivinumOfficium\SetupString.pl`, in the `%predicates`, for the Cistercian version it says `cisterciense`. However, this will result in a gramatically incorrect expression `rubrica cisterciense`, employing the neuter version "cisterciense" with a feminine "rubrica". Could you change it to `cisterciensis`?

I know it's a "Kleinigkeit" and in principle both versions can live together if a duplicate line is added, but still, my classical scholar heart would like the correct version better.

Thanks!

**Reply by FAJ-Munich** [reply]
Wow, that's super interesting. 
Thanks so much for the links. It seems like the Cistercian Breviary has a completely different approach to the Dominica infra 8vam Nativitatis: if it doesn't happen to fall on December 30th, it's simply ignored. And in contrast to the Benedictine 1617 (and partially 1930 books), Romans 1 is not postponed to be read on S. Silvester in this case but S. Silvester simply starts the readings from Romans with Chapter 2.

for the "Kleinigkeit"...  is done! 
I've passed my *Latinum* in school with the worst grade possible thanks to being able to differentiate Daktylus, Jambus, and Trochäus without being able to create a proper translation.
